scrypto/src/component/component.rs

```diff
@@ -372,13 +372,13 @@ impl<O: HasStub> Global<O> {
         }
     }
 
-    fn metadata(&self) -> Attached<Metadata> {
+    pub fn metadata(&self) -> Attached<Metadata> {
```

Comment:
I've tried to keep these private for the most part as `Attached<Metadata>` exposes quite some nastiness underneath. Is there a reason to make these public?